crates/scheduler/src/call/crypto/hash.rs

```diff
@@ -63,9 +64,17 @@ fn initialize<B: Board>(mut call: SchedulerCall<B, api::initialize::Sig>) {
     let result = try {
         let context = match convert_hash_algorithm::<B>(*algorithm)?? {
             #[cfg(feature = "board-api-crypto-sha256")]
-            Algorithm::Sha256 => HashContext::Sha256(board::crypto::Sha256::<B>::default()),
+            Algorithm::Sha256 => {
+                let default = board::crypto::Sha256::<B>::default();
+                default.last_error()?;
```

Comment:
@ia0 You mentioned we could use `Result::map` for this. Could you explain more? Thanks.

Reply:
The `let context =` needs to become `let (context, error) =` and each branch needs to return both, e.g.:

```rust
let context = board::crypto::Sha256::<B>::default();
let error = context.last_error();
(context, error)
```

Then the final `Ok(scheduler.applet.hashes.insert(context)? as u32)` _could_ become:

```rust
error.map(|()| scheduler.applet.hashes.insert(context)? as u32)
```

The problem is that we capture the `?` in the closure, while it should return to the try-block. So this actually doesn't work. We need to write the map manually instead:

```rust
match error {
    Ok(()) => Ok(scheduler.applet.hashes.insert(context)? as u32),
    Err(error) => Err(error),
}
```